File: Design-Patterns/Bridge/Cpp/default_spell.cpp
#include <iostream>
#include <string>
#include "spell.h"

default_spell::default_spell(int power, int speed, int cost) : spell(power, speed, cost)
{

}

void default_spell::cast()
{
	std::cout << "Casts a default air spell with power " + std::to_string(power) + " and speed " + std::to_string(speed) + "." << std::endl;
}

File: Design-Patterns/Bridge/CSharp/Bridge/Spells/DefaultSpell.cs
using System;

namespace Bridge.Spells
{
    public class DefaultSpell : Spell
    {
        public DefaultSpell(int power, int speed, int cost) : base(power, speed, cost)
        {
        }

        public override void Cast()
        {
            Console.WriteLine($"Casts a default air spell with power {Power} and speed {Speed}.");
        }
    }
}
